qiskit/transpiler/passes/optimization/commutation_analysis.py

```diff
@@ -87,6 +87,7 @@ def run(self, dag):
 
 
 def _commute(node1, node2, cache):
+    from qiskit.quantum_info.operators.predicates import is_diagonal_matrix
```

Comment:
Is there a reason to do this at runtime (like a circular dependency)? If not can we do this at the top so it's done at import time.